app/assets/javascripts/dances/new.js

```diff
@@ -0,0 +1,47 @@
+class NewDance{
+  constructor(json) {
+    this.id = json[0].id;
+    this.directions = [];
+  }
+}
+
+dividNewDirections = (dds, newDan) => {
+  dds.each((dd) => {
+    newDan.directions.push(submitDDAdd(dd, newDan));
+  });
+};
```

Comment:
Ah I see you have some of the functionality I mentioned above! Good!

I would define this function on the NewDance class, instead of accepting a newDance object as an argument.